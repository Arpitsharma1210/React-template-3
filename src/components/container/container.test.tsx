import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Container from './index';
import { Provider } from 'react-redux';
import store from '../../redux/store';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/',
  }),
}));

describe('Container component', () => {
  it('renders with default props and correct layout', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Container />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
