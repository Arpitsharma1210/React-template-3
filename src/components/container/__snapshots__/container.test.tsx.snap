// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component renders with default props and correct layout 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg sc-bcHOTY fJiLvO css-1oqqzyl-MuiContainer-root"
>
  <div
    className="sc-gsMHZj gCMIax"
  >
    <div
      className="sc-dkHyXG dcNAHd"
    >
      <img
        className="sc-hLljAB jSMrfv"
        src="/assets/images/smallLogo.png"
      />
    </div>
    <div
      className="sc-eDgayY jLaUws"
    >
      <div
        active={false}
        className="sc-jTkTEj kRlOPU"
        onClick={[Function]}
      >
        <p
          className="MuiTypography-root MuiTypography-body2 sc-gLfKCG dKIwTK css-e784if-MuiTypography-root"
        >
          Dashboard
        </p>
      </div>
    </div>
    <div
      className="sc-iBIYEh gvAxhF"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body2 sc-ftDPCE jnNHKQ css-e784if-MuiTypography-root"
      >
        Logout
      </p>
    </div>
  </div>
</div>
`;
